src/promptflow/promptflow/_sdk/_submitter/experiment_orchestrator.py

```diff
@@ -127,91 +158,542 @@ def _test_flow_node(self, node, test_context):
+            :type completed_node: str
+            :param next_executable_nodes: Executable node list.
+            :type next_executable_nodes: list
+            """
+            if completed_node:
+                # Update node edge mapping in current experiment run.
+                # Remove the edge of the node that has been completed.
+                for referenced_nodes in node_edges_mapping.values():
+                    referenced_nodes.discard(completed_node)
+            next_executable_nodes = [
+                self._nodes[node_name] for node_name, edges in node_edges_mapping.items() if len(edges) == 0
+            ]
+            for node in next_executable_nodes:
+                node_edges_mapping.pop(node.name)
+            return next_executable_nodes
+
+        def check_in_degree_node_outputs(pre_nodes):
+            """Check the input data of nodes already exists, it not return false."""
+            node_runs = {
+                node_name: next(filter(lambda x: x["status"] == ExperimentNodeRunStatus.COMPLETED, node_runs), None)
+                for node_name, node_runs in self.experiment.node_runs.items()
+            }
+            is_in_degree_nodes_ready = True
+            for in_degree_node in pre_nodes:
+                is_in_degree_nodes_ready = in_degree_node in node_runs
+                if node_runs.get(in_degree_node, None):
+                    node_run_info = self.run_operations.get(node_runs[in_degree_node]["name"])
+                    self._node_runs[in_degree_node] = node_run_info
+
+                    output_path = node_run_info.properties.get("output_path", None)
+                    is_in_degree_nodes_ready = is_in_degree_nodes_ready and Path(output_path).exists()
+                else:
+                    is_in_degree_nodes_ready = False
+                    logger.warning(f"Cannot find the outputs of {in_degree_node}")
+            return is_in_degree_nodes_ready
+
+        def stop_process():
+            """
+            Post process of stop experiment. It will update status of all running node to canceled.
+            And update status of experiment to terminated. Then terminate the orchestrator process.
+            """
+            executor.shutdown(wait=False)
+            for future, node in future_to_node_run.items():
+                if future.running():
+                    # Update status of running nodes to canceled.
+                    node.update_exp_run_node(status=ExperimentNodeRunStatus.CANCELED)
+                    self.experiment._append_node_run(node.node.name, ORMRunInfo.get(node.name))
+            # Update status experiment to terminated.
+            self._update_orchestrator_record(status=ExperimentStatus.TERMINATED)
+            # Terminate orchestrator process.
+            sys.exit(1)
+
+        if platform.system() == "Windows":
+            import threading
+
+            def stop_handler():
```

Comment:
extract some util function to another file to make this file shorter

Reply:
encapsulate all the win32 api & psutil apis in this file

Reply:
extract operation about process in utils.pyt